packages/react/error-boundary/src/ErrorBoundaryGroup.tsx

```diff
@@ -0,0 +1,51 @@
+import { ComponentType, createContext, ReactNode, useContext, useEffect, useRef } from 'react';
+import { useIsMounted, useKey } from './hooks';
+
+const ErrorBoundaryGroupContext = createContext({ groupResetKey: {}, resetGroup: () => {} });
+if (process.env.NODE_ENV !== 'production') {
+  ErrorBoundaryGroupContext.displayName = 'ErrorBoundaryGroupContext';
+}
+
+const ErrorBoundaryGroupReset = ({ trigger }: { trigger: ComponentType<{ resetGroup: () => void }> }) => {
+  const { resetGroup } = useErrorBoundaryGroup();
+  const Trigger = trigger;
+
+  return <Trigger resetGroup={resetGroup} />;
+};
+
+export const ErrorBoundaryGroup = ({
+  blockOutside = false,
+  children,
+}: {
+  blockOutside?: boolean;
+  children?: ReactNode;
+}) => {
+  const blockOutsideRef = useRef(blockOutside);
+  const isMounted = useIsMounted();
+  const { groupResetKey } = useErrorBoundaryGroup();
+  const [resetKey, reset] = useKey();
+
+  useEffect(() => {
+    if (isMounted && !blockOutsideRef.current) {
+      reset();
+    }
+  }, [groupResetKey, isMounted, reset]);
+
+  return (
+    <ErrorBoundaryGroupContext.Provider value={{ resetGroup: () => reset, groupResetKey: resetKey }}>
+      {children}
+    </ErrorBoundaryGroupContext.Provider>
+  );
+};
+ErrorBoundaryGroup.Reset = ErrorBoundaryGroupReset;
```

Comment:
If implementing the component is easy and straightforward, I prefer to provide smaller sets of API, since it would improve our maintainability. Users of our library could also benefit in increased flexibility (they could decide how to design `ErrorBoundary.Reset`'s API.) What do you think?

Reply:
I understood your intention. I removed ErrorBoundary.Reset in [b12ec9c](https://github.com/toss/slash/pull/157/commits/b12ec9c8f6418e99582bd037e8d489a1037991c5)